mixins/admin.py

```diff
@@ -0,0 +1,58 @@
+import csv
+
+from django.contrib import admin
+from django.http import HttpResponse
+from django.utils import timezone
+from django.utils.text import slugify
+
+
+class CSVAdminMixin(admin.ModelAdmin):
+    """
+    Allows us to add a csv exporter to admins
+    """
+
+    def get_actions(self, request):
+        actions = self.actions if hasattr(self, "actions") else []
+        actions.append("csv_export")
+        actions = super().get_actions(request)
+        return actions
+
+    def get_export_fields(self, request):
+        """
+        Build a list of model fields to add to the csv
+        """
+        model_fields = self.get_form(request, obj=self).Meta.fields
```

Comment:
Is there a particular reason for taking the fields from the form rather than the model?
I assume there could potentially be a situation where you want to export something that's not shown in the admin

Reply:
This was a bit of a pain. Initially I had it as either a set list, or like the other exporter using `self.Model._meta.fields` but the model meta didn't return a list of strings. Which meant I had to account for that later on where I build the headers/rows. The `get_form` method actually returns a model form so I'm pretty confident this is a kind of midpoint solution

Reply:
I agree that the get_form() would also observe any custom form that maybe be declared, assuming that's a good thing.
There's also the  
`[f.name for f in self.Model._meta.get_fields()]`

Reply:
> I agree that the get_form() would also observe any custom form that maybe be declared, assuming that's a good thing.
> There's also the
> `[f.name for f in self.Model._meta.get_fields()]`

I also tried this. But for some reason it was including all the related fields (m2m, FK etc.) and even though I was using the docs to explicitly hide them, they still persisted. So I settled on this for now

Reply:
Do we not want to be able to get FK & m2m fields in an export?

Reply:
Sorry I meant reverse relationships, so models/objects that have an FK to the one you're trying to export. We could add it later potentially but I think just keeping it simple for now is better, just allowing for the fields that are explicitly defined on the model is what this achieves